### PR TITLE
chore: Update the runner to fetch necessary info from env vars

### LINKED ISSUE #54: PRs from forked branches don't have access to secrets
### Issue:
- Currently any PR opened from a fork will fail CI tests that depend on environment variables. 
- This is because the repository secrets are not accessible for forked workflows
- See [here](https://stackoverflow.com/questions/61419552/workaround-to-post-comments-from-github-actions-from-forked-repos)

### Solution
- Unclear
- Maybe one of the comments from [this issue](https://stackoverflow.com/questions/61419552/workaround-to-post-comments-from-github-actions-from-forked-repos)

### PR DESCRIPTION
**Pull request by Shabirmean**
#### Description
- Github Actions by default does not allow PRs from forked repos to access Secrets setup in the original runner
- Thus, we have workflows that use repository secrets, then any PRs from external contributors will fail
- We had initially used repository secrets to store 2 environment variables required by the Golang tests `(GOOGLE_APPLICATION_CREDENTIALS and GOOGLE_CLOUD_PROJECT)`
- However, multiple external contributor PRs _(and PRs from some automation bots)_ started failing because those PRs don't have access to these secrets
- The argument in multiple threads to why GH Actions doesn't support this is that - if those values can be accessed by external PRs then they are probably not secrets
- I believe that the two variable that our tests require, also need not be secrets persay; thus I've added them env vars that are sourced before the workflow is run
- I have not directly hardcoded them in the workflow to ensure all envs can be stored in a single place upon adding a new runner.

#### Change summary
- Update the GH Actions runner init script to include instructions to set up initial env vars
- Add sourcing instructions to the env vars file before running the Golang tests
- Add one of the module version upgrade changes to mimic PR #55 

#### Related PRs/Issues
- #55

### CODE REVIEW

File: .github/workflows/ci_any_pr.yaml [hunk 1]
```diff
@@ -86,10 +86,8 @@ jobs:
       - name: Golang unit tests
         timeout-minutes: 20
         env:
-          # these env variables are required by the golang tests
-          GOOGLE_CLOUD_PROJECT: ${{ secrets.GOOGLE_CLOUD_PROJECT }}
-          GOOGLE_APPLICATION_CREDENTIALS: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
           GOLANG_TEST_DIR: ${{ matrix.go-test-dir }}
         run: |
+          source /var/local/gh-runner/env_vars
```
Comment (NimJay) [inline .github/workflows/ci_any_pr.yaml:91]: Was this file  *manually* created (in the GitHub action runner virtual machines)?
If so, do you think we need to update the documentation here: https://github.com/GoogleCloudPlatform/anthos-samples/blob/806db81be21edd48a40149aded6127a2ecc15c01/.github/gh_runner.md?

**Review by NimJay — APPROVED** [review]
This pull-request looks good to me.
I'm just not sure if we need to update docs here: https://github.com/GoogleCloudPlatform/anthos-samples/blob/806db81be21edd48a40149aded6127a2ecc15c01/.github/gh_runner.md

File: .github/workflows/ci_any_pr.yaml (hunk 1 above)
Comment (NimJay) [reply .github/workflows/ci_any_pr.yaml:91]: I also noticed that the value of `GOOGLE_APPLICATION_CREDENTIALS` (in the `/var/local/gh-runner/env_vars` file) in both virtual machines (GitHub Action runners) is pointing to a file. Does this file need to be manually created when creating a new GitHub Action? If so, should we mention that in https://github.com/GoogleCloudPlatform/anthos-samples/blob/806db81be21edd48a40149aded6127a2ecc15c01/.github/gh_runner.md?

File: anthos-bm-gcp-terraform/main.tf [hunk 2]
```diff
@@ -55,15 +55,15 @@ locals {
 
 module "enable_google_apis_primary" {
   source                      = "terraform-google-modules/project-factory/google//modules/project_services"
-  version                     = "10.3.2"
+  version                     = "11.1.0"
```
[the remark this answers is not in the document]
Comment (Shabirmean) [reply anthos-bm-gcp-terraform/main.tf:58]: Right. I didnt think this through. You are correct; mimic-ing that PR here doesnt make sense

File: .github/workflows/ci_any_pr.yaml (hunk 1 above)
Comment (Shabirmean) [reply .github/workflows/ci_any_pr.yaml:91]: @NimJay - Yes the files were manually created this time. How they would be created when a new runner is added is by executing [these instructions](https://github.com/GoogleCloudPlatform/anthos-samples/blob/806db81be21edd48a40149aded6127a2ecc15c01/.github/gh_runner_dependencies.sh#L44-L53) **printed** at the end of the `gh_runner_dependencies.sh` script. I'll update the doc to add a note about completing the steps printed at the end of the script.

Yes, it points to the `Service Account` key file from the project. Creation of this file in a new runner is also part of the instructions at the end of the  `gh_runner_dependencies.sh` script

File: anthos-bm-gcp-terraform/main.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (NimJay) [reply anthos-bm-gcp-terraform/main.tf]: Okay, got it! I don't think it hurts to just keep the commit in this PR and merge as is. No action required.

File: .github/workflows/ci_any_pr.yaml (hunk 1 above)
Comment (NimJay) [reply .github/workflows/ci_any_pr.yaml:91]: Ohhh, I didn't realize that the instructions contain a step about `gh_runner_dependencies.sh`!
Nonetheless, the new message is perfect!
Thanks, Shabir!

The pull-request looks good to me. Resolving this thread.

**Pull request merged**